File: akkaserver/src/main/scala/com/lightbend/modelServer/model/ModelFactory.scala
package com.lightbend.modelServer.model

import com.lightbend.modelServer.ModelToServe

/**
 * Created by boris on 5/9/17.
 * Basic trait for model factory
 */
trait ModelFactory {
  def create(input: ModelToServe): Option[Model]
  def restore(bytes: Array[Byte]): Model
}

File: model/src/main/java/com/lightbend/java/model/ModelFactory.java
package com.lightbend.java.model;

import java.util.Optional;

/* Created by boris on 7/14/17. */

/**
 * Basic trait for a model factory.
 */
public interface ModelFactory {
    Optional<Model> create(CurrentModelDescriptor descriptor);
    Model restore(byte[] bytes);
}
